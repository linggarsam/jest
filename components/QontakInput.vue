<template>
  <b-input-group>
    <b-form-input
      autocomplete="off"
      :value="value"
      @input="onInput"
    />
  </b-input-group>
</template>

<script>
export default {
  props: {
    value: {
      type: String,
      default: () => ''
    }
  },
  methods: {
    onInput (val) {
      this.$emit('input', val)
    }
  }
}
</script>
