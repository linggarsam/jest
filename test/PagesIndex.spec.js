import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import PagesIndex from '@/pages/index.vue'
import QontakInput from '@/components/QontakInput.vue'
import * as BootstrapVue from 'bootstrap-vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

test('Test mounted', () => {
  const wrapper = mount(PagesIndex, {
    localVue,
    components: {
      QontakInput
    }
  })
  expect(wrapper.vm).toBeTruthy()
});

test('Test snapshot', () => {
  const wrapper = mount(PagesIndex, {
    localVue,
    components: {
      QontakInput
    }
  })
  expect(wrapper.html()).toMatchSnapshot()
});

test('Test checkAnswer sudah terjawab method', () => {
  const wrapper = shallowMount(PagesIndex, {
    localVue,
    components: {
      QontakInput
    }
  })
  wrapper.vm.userAnswer = 'a'
  wrapper.vm.checkAnswer()
  expect(wrapper.vm.result).toBe('Sudah terjawab')
})

test('Test checkAnswer tidak terjawab method', () => {
  const wrapper = shallowMount(PagesIndex, {
    localVue,
    components: {
      QontakInput
    }
  })
  wrapper.vm.userAnswer = ''
  wrapper.vm.checkAnswer()
  expect(wrapper.vm.result).toBe('Belum terjawab')
})

test('Test checkAnswer tidak sesuai pilihan method', () => {
  const wrapper = shallowMount(PagesIndex, {
    localVue,
    components: {
      QontakInput
    }
  })
  wrapper.vm.userAnswer = 'c'
  wrapper.vm.checkAnswer()
  expect(wrapper.vm.result).toBe('Jawaban tidak sesuai pilihan')
})