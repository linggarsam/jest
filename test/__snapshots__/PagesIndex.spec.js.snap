// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
"<div style=\\"width: 20%; padding: 5px;\\">
  <div>
    Pertanyaan 1
  </div>
  <div><label>a. Ya</label></div>
  <div><label>b. Tidak</label></div>
  <div role=\\"group\\" class=\\"input-group\\">
    <!----><input type=\\"text\\" autocomplete=\\"off\\" class=\\"form-control\\">
    <!---->
  </div>
  <div><label>
      Belum terjawab
    </label></div>
</div>"
`;
